---
author: Lisa Hu
output:
  pdf_document:
    includes:
      before_body: title.sty
    toc: true
    toc_depth: 2
    number_sections: true
    latex_engine: xelatex
---

[//]: # (Page 1, toc)

\newpage

[//]: # (Page 2)

# Loading the Data
```{r setup, include = TRUE, warning = FALSE, message = FALSE, results = "hide"}
#' Setup chunk
knitr::opts_chunk$set(cache = TRUE)
knitr::opts_chunk$set(echo = TRUE)

# Load packages
packages <- c("pander", "dplyr", "affy", "knitr", "ggplot2", "DESeq2", "pheatmap",
              "PoiClaClu", "scales", "apeglm", "EnhancedVolcano", "crayon")
invisible(lapply(packages, library, character.only = T))
require(httr)
require(jsonlite)
```

For decompressing the data^[https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129479], run the code chunks in the Rmd file that deem fit for your situation:

* If you downloaded the data from the official site: Decompress the data and run the Rscript `data_loading.R`.
* If you want to use the dataset delivered with the project: Run the `decompress-dataset` code chunk.


```{r decompress-dataset, eval = FALSE}
#' Decompress the complete dataset
#' Use this chunk if you did not download the data from the site and want to use
#'  the delivered gzipped dataset

## Set the count.file variable to the full path of the gene file
count.file <- ""
system(paste("gzip -d", count.file))
```

After decompressing the data, the data can be read:
```{r load-dataset}
## Read the dataset
dataset <- read.table("./gene_count.txt", sep = "\t", header = TRUE)
## Set rownames of the dataset to first column
row.names(dataset) <- dataset$Gene
## Remove the Gene column
dataset <- dataset[-1]
## Remove the first 5 rows (these rows are not genes)
dataset <- dataset[!(rownames(dataset) %in% c("__no_feature", "__ambiguous",
                     "__alignment_not_unique", "__too_low_aQual", "__not_aligned")),]

## Column indices for dataset
glucose.data <- seq(1, 48, 2)
galactose.data <- seq(2, 49, 2)
groups <- factor(rep(1:2, times=24), labels = c("Glucose", "Galactose"))
col.ordered <- c(colnames(dataset[glucose.data]), colnames(dataset[galactose.data]))

## Colors for the two sample groups (red = galactose, blue = glucose)
group.cols <- hue_pal()(2)
```

\newpage

[//]: # (Page 3)

# Exploratory Data Analysis
## Data Sample
```{r data-sample}
pander(dataset[0:5, 0:4], split.tables = 64)
pander(summary(dataset[,0:6]), split.tables = 64)
```

\newpage

[//]: # (Page 4)

## Boxplots
```{r boxplots}
layout(matrix(c(1,1,1,2,2,2), nrow = 6, ncol = 1, byrow = T))
## Glucose plot
boxplot(log2(dataset[glucose.data]+1), main = "Glucose", names = seq(1, 24),
        xlab = "Sample", ylab = "log2(Count)", outline = FALSE)
## Galactose plot
boxplot(log2(dataset[galactose.data]+1), main = "Galactose", names = seq(1, 24),
        xlab = "Sample", ylab = "log2(Count)", outline = FALSE)
```

\newpage

[//]: # (Page 5)
## Density Plots
```{r density plot}
plotDensity(log2(dataset+1), main = "Density plot", col = group.cols,
            lty = 1:48, xlab = "log2(count)")
legend("topright", names(dataset), lty = 1:48, col = group.cols,
       cex = 0.7, ncol = 3)
```

\newpage

[//]: # (Page 6)
## Barplots
```{r sequencing-depth, fig.height = 6}
layout(matrix(c(1,1,1,2,2,2), nrow = 6, ncol = 1, byrow = T))
## Barplot of first half of the data
x1 <- barplot(colSums(dataset[1:24]/ 1e6), main = "Sequencing depth sample 1-12",
              xlab = "Sample", ylab = expression("Sequencing depth (x10"^6*")"),
              ylim = c(0, 70), las = 2, col = group.cols, xaxt = 'n')
text(x = x1, y = colSums(dataset[1:24]/ 1e6),
     label = round(colSums(dataset[1:24]/ 1e6),0), pos = 3)
axis(1, at = x1, labels = rep(1:12, each = 2), tick = FALSE, cex = 0.6)
legend("topright", c("Glucose", "Galactose"), col = group.cols, pch = 19)

## Rest of the data
x2 <- barplot(colSums(dataset[25:48]/ 1e6), main = "Sequencing depth sample 13-24",
              xlab = "Sample", ylab = expression("Sequencing depth (x10"^6*")"),
              ylim = c(0, 60), las = 2, col = group.cols, xaxt = 'n')
text(x = x2, y = colSums(dataset[25:48]/ 1e6),
     label = round(colSums(dataset[25:48]/ 1e6), 0), pos = 3)
axis(1, at = x1, labels = rep(13:24, each = 2), tick = FALSE, cex = 0.6)
```

\newpage

[//]: # (Page 7)

# Normalization
```{r normalization, message = FALSE, warning = FALSE}
ddsMat <- DESeqDataSetFromMatrix(countData = round(dataset),
                                 colData = data.frame(samples = names(dataset)),
                                                      design = ~ 1)
rld.dds <- vst(ddsMat)
rld <- assay(rld.dds)
sampledists <- dist(t(rld))
```

## Heatmaps
```{r heatmap-original}
distMatrix <- as.matrix(sampledists)

annotation <- data.frame(GrowthMedium = groups)

rownames(annotation) <- names(dataset)

pheatmap(distMatrix, show_colnames = T,
         annotation_col = annotation,
         clustering_distance_rows = sampledists,
         clustering_distance_cols = sampledists,
         main = "Euclidean Sample Distances", fontsize= 6)
```
\newpage

[//]: # (Page 8)

A plot without the clustering and ordered groups:
```{r heatmap-clusterless}
rld.ord <- rld[,col.ordered]
sampledists.ord <- dist(t(rld.ord))

distMatrix.ord <- as.matrix(sampledists.ord)

annotation.ord <- data.frame(GrowthMedium = factor(rep(1:2, each = 24),
                                               labels = c("Glucose", "Galactose")))

rownames(annotation.ord) <- col.ordered

pheatmap(distMatrix.ord, show_colnames = TRUE,
         annotation_col = annotation.ord, cluster_rows = FALSE, cluster_cols = FALSE,
         main = "Euclidean Sample Distances", fontsize= 6)
```
\newpage

[//]: # (Page 9)
## Multi Dimensional Scaling
Since the colours confirm the sample group, the sample names were simplified to only the number indicating the different samples.
```{r mds}
dds <- assay(ddsMat)
poisd <- PoissonDistance(t(dds), type="deseq")
## Extract matrix with distances
poisDistMatrix <- as.matrix(poisd$dd)
## Calculate MDS for X- and Y- coordinates
mdsPoisData <- data.frame(cmdscale(poisDistMatrix))
## Readable names
names(mdsPoisData) <- c("x_coord", "y_coord")
## Annotation label
coldata <- rep(1:24, each=2)

ggplot(mdsPoisData, aes(x_coord, y_coord, color = groups, label = coldata)) +
        geom_text(size = 3) +
        ggtitle("Multi Dimensional Scaling") +
        labs(x = "Poisson Distance", y = "Poisson Distance") +
        theme_bw()
```

\newpage

[//]: # (Page 10)
# Discovering Differentially Expressed Genes (DEGs)
## The Fold Change Value
```{r fold-change}
fpm <- log2( (dataset / (colSums(dataset) / 1e6)) + 1 )

## New columns for average fpm
fpm$avg_glu <- rowMeans(fpm[glucose.data])
fpm$avg_gal <- rowMeans(fpm[galactose.data])

## Calculate FC
fpm$fc_galglu <- fpm$avg_gal - fpm$avg_glu

## Create histogram of the fold changes
hist(fpm$fc_galglu, main = "Fold Change (GAL vs GLU)",
     col=group.cols[2], breaks=100, xlab = "Fold Change")
## Vertical ablines
abline(v = c(1,-1), lty = 2, col = group.cols[1])
```
\newpage

[//]: # (Page 11)
## DESeq2 Analysis
Genes with average read count below 10 and with zero counts in more than 20% of samples (10 samples) were considered not expressed and filtered.
```{r data-filtering, message = FALSE}
## Create design frame
design <- data.frame(groups, row.names = colnames(dataset))
## Create new DDS object with correct design
dds <- DESeqDataSetFromMatrix(countData = dataset,
                                 colData = design, design = ~ groups)
## Keep genes with more than 1 count
dds <- dds[ rowSums(counts(dds)) > 1, ]
dds <- DESeq(dds)
dds

## Results
dds.res <- results(dds, alpha=0.05)

## Filter out reads below 10
dds.filtered <- dds.res[!dds.res$baseMean < 10,]

## Shrinkage
resultsNames(dds)
lfc.gal <- lfcShrink(dds, coef = "groups_Galactose_vs_Glucose", res = dds.res,
                     type = "apeglm")
lfc.gal
summary(lfc.gal)
```

\newpage

[//]: # (Page 12)
### MA Plot
```{r ma-plot}
DESeq2::plotMA(lfc.gal, main = "Glucose vs galactose", ylim = c(-2, 2))
```
\newpage

[//]: # (Page 13)

# Data analysis and Visualization
## Volcano Plot
```{r volcano}
EnhancedVolcano(lfc.gal, x = 'log2FoldChange', y = 'padj', lab=rownames(lfc.gal),
                title = "Glucose vs galactose",
                subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                # Change text and icon sizes
                labSize = 3, pointSize = 1.5, axisLabSize=10, titleLabSize=12,
                subtitleLabSize=8, captionLabSize=10,
                # Disable legend
                legendPosition = "none",
                # Set cutoffs
                pCutoff = 0.05, FCcutoff = 2)

```

The points in red in this plot are important to find out what genes are differentially expressed. To do this, we can subset the data (`lfc.gal`) to filter the red dots:
```{r subset-shrinkage}
genes <- subset(lfc.gal, lfc.gal[,'log2FoldChange'] > 2 & -log10(lfc.gal[,'padj']) > 0.05
                | lfc.gal[,'log2FoldChange'] < -2)
```
\newpage

[//]: # (Page 14)
The rownames are gene IDs from the Ensembl database. Using an Ensembl ID to Gene Symbol converter^[https://www.biotools.fr/human/ensembl_symbol_converter], the gene names can be found:
```{r id-request, message = FALSE}
## Get the IDs
url <- "https://biotools.fr/human/ensembl_symbol_converter/"
ids <- rownames(genes)
ids_json <- toJSON(ids)

## Create the request
body <- list(api = 1, ids = ids_json)
req <- POST(url, body = body)

output <-  fromJSON( content(req, "text"), flatten = TRUE )
df <- data.frame(unlist(output))
colnames(df) <- "Gene symbol"
pander(df, split.tables = 64)
```

\newpage

[//]: # (Page 15, references)
